### DISCUSSION TITLE
Invisible occlusion mesh !

### DISCUSSION THREAD

**Discussion by dthery** [google/model-viewer#2797] [Q&A]
Can we use the colorWrite property of three.js to create an invisible occlusion mesh that hides some areas of a model to make transparency? 
https://threejs.org/docs/#api/en/materials/Material.colorWrite

**Comment by elalish** [comment]
We don't expose that in our API since it's not part of glTF. Of course you can always fork us or hack into our internals to accomplish this, though the usual caveats apply regarding stability. If there is significant interest and use cases for it, we'd be happy to consider adding it, perhaps also to the glTF spec.

**Reply by MathDestruction** [reply]
Pushing my luck here... But no update on this yet? An Occlusion material/property that we could use to hide content behind?

**Reply by elalish** [reply]
@MathDestruction Can you be a bit more specific about your use case? I'd like to make sure we find a solution that actually solves your problem.

**Reply by MathDestruction** [reply]
Of course.

I'm looking use a plane or cube mesh to occlude a 3D model so that it can appear to come out of the ground. The main use case would be a "reveal" animation where after localizing your object, a short animation will play opening up a door where the main object comes out of. At the moment, unless I have missed it, models will be visible under ground plane with nothing to occlude them

I've done similar things in unity using stencil buffers, if that helps at all. And thank you for your fast reply!

**Reply by felipefierro** [reply]
Hey! Trying my luck too but Any news on having matte/ shadow material? Or occluded objects behind certain geometry? Thank you so much!

**Comment by felipefierro** [comment]
> Can we use the colorWrite property of three.js to create an invisible occlusion mesh that hides some areas of a model to make transparency?
> https://threejs.org/docs/#api/en/materials/Material.colorWrite

Hi! did you manage to find a solution?

**Comment by felipefierro** [comment]
Hi! I think that what we all meant is something like this:

A material or property for occlude objects behind certain geometry in order to make them "invisible" 
![hand](https://github.com/google/model-viewer/assets/134464697/8b851a05-1b8c-4e12-9a4b-67d8beefd3c9)

I know it's a topic from 2021 so did you guys manage to figure it out somehow? thank you so much!

**Reply by plushviedev4** [reply]
I Want Exactlyyy this. Have you find anything ??

**Comment by Bettole** [comment]
Hello.. have you known how to handle this? I'm looking for information about it and I haven't been able to find solutions!!
